src/Sulu/Component/Media/SystemCollections/SystemCollectionManager.php

```diff
@@ -0,0 +1,262 @@
+     * @var array
+     */
+    private $systemCollections;
+
+    public function __construct(
+        array $config,
+        CollectionManagerInterface $collectionManager,
+        EntityManagerInterface $entityManager,
+        TokenStorageInterface $tokenProvider = null,
+        ConfigCache $cache,
+        $locale
+    ) {
+        $this->config = $config;
+        $this->collectionManager = $collectionManager;
+        $this->entityManager = $entityManager;
+        $this->tokenProvider = $tokenProvider;
+        $this->cache = $cache;
+        $this->locale = $locale;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function warmUp()
+    {
+        $this->getSystemCollections();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSystemCollection($key)
+    {
+        $systemCollections = $this->getSystemCollections();
+
+        if (!array_key_exists($key, $systemCollections)) {
+            throw new UnrecognizedSystemCollection($key, array_keys($systemCollections));
+        }
+
+        return $systemCollections[$key];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isSystemCollection($id)
+    {
+        return in_array($id, $this->getSystemCollections());
+    }
+
+    /**
+     * Returns system collections.
+     *
+     * @return array
+     */
+    private function getSystemCollections()
+    {
+        if (!$this->systemCollections) {
+            if (!$this->cache->isFresh()) {
+                $systemCollections = $this->buildSystemCollections(
+                    $this->locale,
+                    $this->getUserId()
+                );
+
+                $this->cache->write(serialize($systemCollections));
+            }
+
+            $this->systemCollections = unserialize($this->cache->read());
+        }
+
+        return $this->systemCollections;
+    }
+
+    /**
+     * Returns current user.
+     *
+     * @return int
+     */
+    private function getUserId()
+    {
+        if (!$this->tokenProvider || ($token = $this->tokenProvider->getToken()) === null) {
+            return;
+        }
+
+        return $token->getUser()->getId();
+    }
+
+    /**
+     * Go thru configuration and build all system collections.
+     *
+     * @param string $locale
+     * @param int $userId
+     *
+     * @return array
+     */
+    private function buildSystemCollections($locale, $userId)
+    {
+        $root = $this->getOrCreateRoot('system_collections', 'System', $locale, $userId);
+
+        $collections = ['root' => $root->getId()];
+        foreach ($this->config as $namespaceKey => $namespaceItem) {
+            $namespace = $this->getOrCreateCollection(
+                $namespaceKey,
+                $namespaceItem['meta_title'],
+                $userId,
+                $root->getId()
+            );
+            $collections[$namespaceKey] = $namespace->getId();
+
+            foreach ($namespaceItem['collections'] as $collectionKey => $collectionItem) {
```

Comment:
hmm the config is no able to go deeper but your right the service should be able to ... i will change that